### PR TITLE
[WIP] Bower componentization

### PR DESCRIPTION
**Pull request by adambraimbridge**
DO NOT MERGE @commuterjoy

### CODE REVIEW

**Comment by adambraimbridge** [comment]
@commuterjoy Still [WIP], as the corresponding apps (next-js-setup, homepage, articles, streams) will require simultaneous releases, but ready for an initial review now.

**Comment by adambraimbridge** [comment]
@tom-parker @ifyio : can you please review? I'm available at your convenience if you want to discuss. </a>

**Comment by adambraimbridge** [comment]
TODO: 
- [ ] Update the README as appropriate

**Comment by commuterjoy** [comment]
+1 looks good, merge this in to master

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.PHONY: test
+
+clean:
+	git clean -fxd
+
+install:
+	obt install --verbose
+
+build:
+	obt build --js=src/main.js --sass=src/main.scss --buildFolder=public
```
Comment (commuterjoy) [inline Makefile]: probably don't need the main.scss as there's no CSS being produced yet

File: package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "n-third-party-code",
+  "version": "0.0.0",
+  "description": "Bower component: Load and manage third-party code across next.ft.com",
+  "main": "main.js",
```
Comment (commuterjoy) [inline package.json]: should be `src/main.js`

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+var nextAdsComponent = require("next-ads-component");
+var flagsClient = require('next-feature-flags-client').init();
+
+var ThirdPartyJsSetup = {
+	init:function () {
+
+		// o-tracking
+		flagsClient.then(function(flags) {
+			if (flags.get('oTracking')) {
+				require('./o-tracking');
```
Comment (commuterjoy) [inline src/main.js]: it's a bit confusing having a local file called 'o-tracking' and a local file called './o-tracking'.

... well not that confusing, just requires a second take :)

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var nextAdsComponent = require("next-ads-component");
+var flagsClient = require('next-feature-flags-client').init();
+
+var ThirdPartyJsSetup = {
+	init:function () {
```
Comment (adambraimbridge) [inline src/main.js]: init:function(flags){

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var nextAdsComponent = require("next-ads-component");
+var flagsClient = require('next-feature-flags-client').init();
```
Comment (adambraimbridge) [inline src/main.js]: delete

File: src/main.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+var ThirdPartyJsSetup = {
+	init:function () {
+
+		// o-tracking
+		flagsClient.then(function(flags) {
+			if (flags.get('oTracking')) {
+				require('./o-tracking-wrapper');
+			}
+		});
+
```
Comment (adambraimbridge) [inline src/main.js]: // o-tracking
+			if (flags.get('oTracking')) {
+				require('./o-tracking-wrapper');
+			}

File: src/main.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		// Initialise once page loads
+		window.addEventListener('load', function () {
+			flagsClient.then(function(flags) {
+
```
Comment (adambraimbridge) [inline src/main.js]: delete this flags wrapper

**Pull request merged**